contracts/src/RatesManager.sol

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.28;
+
+import "@openzeppelin/contracts/access/extensions/AccessControlDefaultAdminRules.sol";
+
+contract RatesManager is AccessControlDefaultAdminRules {
+    // Create a dedicated role for managing rates
+    bytes32 public constant RATES_ADMIN_ROLE = keccak256("RATES_ADMIN_ROLE");
+
+    // Rates struct holds the fees and the start time of the rates
+    struct Rates {
+        uint64 messageFee;
+        uint64 storageFee;
+        uint64 congestionFee;
+        uint64 startTime;
+    }
+
+    // All Rates appended here
+    Rates[] private allRates;
+
+    // Event emitted when new Rates are added
+    event RatesAdded(uint64 messageFee, uint64 storageFee, uint64 congestionFee, uint64 startTime);
+
+    constructor()
+        AccessControlDefaultAdminRules(
+            1 days, // adminTransferDelay
+            msg.sender // initialAdmin
+        )
+    {
+        // Setup RATES_ADMIN_ROLE so that the default admin can grant it
+        _setRoleAdmin(RATES_ADMIN_ROLE, DEFAULT_ADMIN_ROLE);
+
+        // Grant the deployer the RATES_ADMIN_ROLE
+        _grantRole(RATES_ADMIN_ROLE, msg.sender);
+    }
+
+    /**
+     * @dev Add new Rates. Can only be called by addresses with RATES_ADMIN_ROLE.
+     *      The array only grows; we do not allow removal or updating.
+     */
+    function addRates(uint64 messageFee, uint64 storageFee, uint64 congestionFee, uint64 startTime)
```

Comment:
Is there any legal reason/concern regarding having lower/upper bounds for fees?

Reply:
Maybe. I'm waiting for more guidance on that, and won't put any bounds in until I'm sure it's necessary.